src/back/spv/block.rs

```diff
@@ -933,10 +933,40 @@ impl<'w> BlockContext<'w> {
                 reject,
             } => {
                 let id = self.gen_id();
-                let condition_id = self.cached(condition);
+                let mut condition_id = self.cached(condition);
                 let accept_id = self.cached(accept);
                 let reject_id = self.cached(reject);
 
+                let condition_ty = self.fun_info[condition]
+                    .ty
+                    .inner_with(&self.ir_module.types);
+                let object_ty = self.fun_info[accept].ty.inner_with(&self.ir_module.types);
+
+                if let (
+                    &crate::TypeInner::Scalar { kind, width },
```

Comment:
This should always check for booleans in the kind: `kind: crate::ScalarKind::Bool`